django/apps/aggregated/management/commands/update_aggregated_data.py

```diff
@@ -23,13 +23,15 @@
     )
     (
         -- Retrieve used tasks
+        -- task_id is not the primary key. project_id-group_id-task_id is
         WITH used_tasks as (
             SELECT
-              project_id, group_id, task_id
-            FROM results R
-              INNER JOIN tasks T USING (project_id, group_id, task_id)
+              MS.project_id, MS.group_id, MSR.task_id
+            FROM mapping_sessions_results MSR
+                INNER JOIN mapping_sessions MS USING (mapping_session_id)
+                INNER JOIN tasks T USING (project_id, group_id, task_id)
             WHERE
-                R.timestamp >= %(from_date)s and R.timestamp < %(until_date)s
+                MS.start_time >= %(from_date)s and MS.start_time < %(until_date)s
             GROUP BY project_id, group_id, task_id
```

Comment:
so we would not aggregate by task anymore. But I think this is not needed anyway as you are aggregating per mapping session (project_id, group_id, user_id) during the next step.

Reply:
Sure, after confirmation in Related to https://github.com/mapswipe/python-mapswipe-workers/pull/607/files/f7ef317bb36b2736dfb7d4471acb9606831bde18#diff-cb03d32601d78f2f4e96b346f6c0dcae523d8301adb4a1e4ae35838a73def35a I will make the change.

Reply:
I have pushed recommended changes.